modules/auxiliary/cloud/aws/enum_ssm.rb

```diff
@@ -0,0 +1,154 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'aws-sdk-ssm'
+require 'aws-sdk-ec2'
+
+class MetasploitModule < Msf::Auxiliary
+  include Rex::Proto::Http::WebSocket::AmazonSsm
+  include Msf::Auxiliary::CommandShell
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Amazon Web Services EC2 instance enumeration',
+        'Description' => %q(
+                          Provided AWS credentials, this module will call the authenticated
+                          API of Amazon Web Services to list all SSM-enabled EC2 instances
+                          accessible to the account. Once enumerated as SSM-enabled, the
+                          instances can be controlled using out-of-band WebSocket sessions
+                          provided by the AWS API (nominally, privileged out of the box).
+                          This module provides not only the API enumeration identifying EC2
+                          instances accessible via SSM with given credentials, but enables
+                          session initiation for all identified targets (without requiring
+                          target-level credentials). The module also provides an EC2 ID
+                          filter and a limiting throttle to prevent session stampedes or
+                          expensive messes.
+                         ),
+        'Author'      => [
+          'Aaron Soto <aaron.soto@rapid7.com>', # EC2 enum module
+          'RageLtMan <rageltman[at]sempervictus>' # SSM stuff
+        ],
+        'License'     => MSF_LICENSE
+      )
+    )
+
+    register_options(
+      [
+        OptBool.new('GET_SSM_SESSION', [true, 'Automatically get SSM sessions once found', false]),
```

Comment:
So because you're including `Msf::Auxiliary::CommandShell`, which includes `Msf::Sessions::CommandShellOptions`, you already have a `CreateSession` datastore option that "Create a new session for every successful login".

It'd make sense to use that datastore option name, even if the description is updated since even though you're logging in, this isn't a login scanner?

Reply:
Roger, wilco